src/Apps/Order/Components/AddressVerificationFlow.tsx

```diff
@@ -195,90 +114,114 @@ const AddressVerificationFlow: React.FC<AddressVerificationFlowProps> = ({
+            key: `suggestedAddress-${index}`,
+            recommended: index === 0,
+            lines: lines,
+            address: address,
+          }))
+        addressOptions = [...suggestedOptions, inputOption]
       } else {
-        if (verificationStatus === "VERIFIED_WITH_CHANGES") {
-          setModalType(ModalType.SUGGESTIONS)
-          trackEvent({
-            action_type: "validationAddressViewed",
-            context_module: ContextModule.ordersShipping,
-            context_page_owner_type: OwnerType.ordersShipping,
-            context_page_owner_id: contextPageOwnerSlug,
-            user_id: userId,
-            flow: "user adding shipping address",
-            subject: "Confirm your delivery address",
-            option: "suggestions",
-          })
-
-          const suggestedOptions = (suggestedAddresses || [])
-            .slice(0, 1)
-            .map(({ address, lines }: any, index) => ({
-              key: `suggestedAddress-${index}`,
-              recommended: index === 0,
-              lines: lines,
-              address: address,
-            }))
-          setAddressOptions([...suggestedOptions, inputOption])
-        } else {
-          setAddressOptions([inputOption])
-          setModalType(ModalType.REVIEW_AND_CONFIRM)
-          trackEvent({
-            action_type: "validationAddressViewed",
-            context_module: ContextModule.ordersShipping,
-            context_page_owner_type: OwnerType.ordersShipping,
-            context_page_owner_id: contextPageOwnerSlug,
-            user_id: userId,
-            flow: "user adding shipping address",
-            subject: "Check your delivery address",
-            option: "review and confirm",
-          })
-        }
+        addressOptions = [inputOption]
+        verificationPath = VerificationPath.REVIEW_AND_CONFIRM
       }
     }
-  }, [
-    inputAddress,
-    onChosenAddress,
-    suggestedAddresses,
-    verificationStatus,
-    trackEvent,
-    userId,
-    contextPageOwnerSlug,
-    hasError,
-    verificationInput,
-  ])
-
-  if (verificationStatus === "VERIFIED_NO_CHANGE" || hasError)
-    return <div data-testid="emptyAddressVerification"></div>
+  }
 
-  if (!modalType || addressOptions.length === 0) return null
+  const modalTitle = modalTitles[verificationPath]
 
-  return (
+  const [selectedAddressKey, setSelectedAddressKey] = useState<
+    AddressOptionKey | undefined
+  >(addressOptions[0]?.key)
+
+  // perform only once when the flow first loads
+  useEffect(() => {
+    if (verificationPath === VerificationPath.ERROR_IMMEDIATE_CONFIRM) {
+      const fallbackOption = fallbackFromFormValues(verificationInput)
+      onChosenAddress(AddressVerifiedBy.USER, fallbackOption.address)
+    } else if (verificationPath === VerificationPath.IMMEDIATE_CONFIRM) {
+      onChosenAddress(AddressVerifiedBy.ARTSY, addressOptions[0].address!)
+    } else {
+      setShowModal(true)
+
+      trackViewedModal({
```

Comment:
Question: firing the viewed event right after setting a state seems to be a few steps early than the modal is rendered to the user. Is this generally safe?

Reply:
If we come back to this and want to add values that are not present immediately on render (eg tracking a value that is null to start but then changed with a `set_` we would need the additional `useRef` that I removed in that commit 👆 . Otherwise this is the right way to do it according to web-practice.

Reply:
Yeah I wasn't sure if there will be cases that modal is not actually rendered, or more impression tracking fired than necessary when react batches state update. Sounds like it's the recommended way. 👍 

Linking to the [discussion](https://artsy.slack.com/archives/C2TQ4PT8R/p1693247057861649) for completeness.